sdk/storage/azure-storage-file-datalake/azure/storage/file/out/datalake/_data_lake_service_client.pyi

```diff
@@ -0,0 +1,24 @@
+# Stubs for datalake._data_lake_service_client (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from ._file_system_client import FileSystemClient
+from ._models import FileSystemPropertiesPaged
+from ._serialize import convert_dfs_url_to_blob_url
+from azure.core.paging import ItemPaged
+from azure.storage.blob._shared.base_client import StorageAccountHostsMixin
+from azure.storage.file.datalake import DataLakeDirectoryClient, DataLakeFileClient
+from azure.storage.file.datalake._models import UserDelegationKey
+from typing import Any, Optional
+
+class DataLakeServiceClient(StorageAccountHostsMixin):
+    def __init__(self, account_url: str, credential: Optional[Any]=..., **kwargs: Any) -> Any: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, credential: Optional[Any]=..., **kwargs: Any) -> DataLakeServiceClient: ...
+    def get_user_delegation_key(self, key_start_time: datetime, key_expiry_time: datetime, **kwargs: Any) -> UserDelegationKey: ...
```

Comment:
Are both start and expiry time required parameters?

Reply:
yeah they are both required according to rest api https://docs.microsoft.com/en-us/rest/api/storageservices/get-user-delegation-key#request-body

Reply:
fix blob api and make key_start_time optional